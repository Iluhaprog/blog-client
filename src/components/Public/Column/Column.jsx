import styled from 'styled-components';
import PropTypes from 'prop-types';

const Column = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: ${({justifyContent}) => justifyContent};
  align-items: ${({alignItems}) => alignItems};
`;

Column.propTypes = {
  justifyContent: PropTypes.oneOf([
    'flex-start',
    'flex-end',
    'center',
    'space-between',
    'space-around',
  ]),
  alignItems: PropTypes.oneOf([
    'flex-start',
    'flex-end',
    'start',
    'end',
    'center',
    'stretch',
  ]),
};

export {Column};
