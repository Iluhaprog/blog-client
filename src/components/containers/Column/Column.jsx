import React from 'react';

export default ({ children, justifyContent, alignItems, wrap }) => {
    const jc = justifyContent ? `column_jc-${justifyContent}` : '';
    const ai = alignItems ? `column_ai-${alignItems}` : ''
    const w = wrap ? `column_w-${wrap}` : ''; 
    return (
        <div className={`column ${jc} ${ai} ${w}`}>
            { children }
        </div>
    );
}