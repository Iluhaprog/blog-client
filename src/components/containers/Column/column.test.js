import React from 'react';
import { render } from '@testing-library/react';
import Column from './Column';

test('Test column row component (default view)', () => {
    const { asFragment } = render(<Column></Column>);
    expect(asFragment()).toMatchSnapshot();
});

test('Test of column component  (view witn jystify-content and align-items)', () => {
    const { asFragment } = render(
        <Column
            justifyContent='c'
            alignItems='c'   
            wrap='w'
        >
        </Column>
    );
    expect(asFragment()).toMatchSnapshot();
});