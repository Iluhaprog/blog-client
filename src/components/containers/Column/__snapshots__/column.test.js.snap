// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test column row component (default view) 1`] = `
<DocumentFragment>
  <div
    class="column   "
  />
</DocumentFragment>
`;

exports[`Test of column component  (view witn jystify-content and align-items) 1`] = `
<DocumentFragment>
  <div
    class="column column_jc-c column_ai-c column_w-w"
  />
</DocumentFragment>
`;
